### PR TITLE
FlyBuy SDK Module Separation into each package

### PR DESCRIPTION
**Pull request by addingama**
This is the PR for the module separation. I've deleted all the old codes except `docusaurus-docs` and `example-app`

~~`example-app` need to be re-write to use the latest package~~

[](https://github.com/BILDIT-Platform/react-native-bildit-flybuy/pull/88)

~~`docusaurus-docs` needs to be adjusted to use the latest package~~ There are parts of docs that need to be reviewed 

The packages will be released under `@bildit-platform/reac-native-flybuy-*` where `*` can be `core`, `pickup`, `presence`, `notify`, and `livestatus`

### CODE REVIEW

**Review by garriott-bildit — APPROVED** [review]

File: docusaurus-docs/docs/Migration.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+```js
+  import * as FlyBuyCore from 'react-native-bildit-flybuy-core';
+  import * as FlyBuyPickup from 'react-native-bildit-flybuy-pickup';
+  import * as FlyBuyNotify from 'react-native-bildit-flybuy-notify';
+  import * as FlyBuyPresence from 'react-native-bildit-flybuy-presence';
+  import * as FlyBuyLiveStatus from 'react-native-bildit-flybuy-livestatus';
+```
+
+All function names are not modified, you just need to change `FlyBuy.<MODULENAME>` to one of the above import alias.
```
Comment (botnerd) [inline docusaurus-docs/docs/Migration.md:38]:
```suggestion
Version 2.20.x and higher move to a modularized architecture where the react native modules mirror the modules in the native SDK. This allows app to pull in only the modules they need. To migrate to the new architecture, please perform the following steps.

## package.json dependencies

Remove `"react-native-bildit-flybuy"` from package.json dependencies, and include the modules that are needed as follows. Core is required. Other modules depend on the features your app is using.

```json
  "dependencies": {
    "react-native-bildit-flybuy-core": "2.20.4",
    "react-native-bildit-flybuy-livestatus": "2.20.4",
    "react-native-bildit-flybuy-notify": "2.20.4",
    "react-native-bildit-flybuy-pickup": "2.20.4",
    "react-native-bildit-flybuy-presence": "2.20.4"
  }
```

## Update code

Most of the code remains the same; however, new imports are needed and the main core class has changed.

Version 2.3.x

```js
  import FlyBuy from 'react-native-bildit-flybuy';

  FlyBuy.Core.Orders.fetchOrders()
      .then(orders => console.tron.log('orders', orders))
      .catch(err => console.tron.log(err));
```

Version 2.20.x

```js
  import * as FlyBuyCore from 'react-native-bildit-flybuy-core';

  FlyBuyCore.Orders.fetchOrders()
      .then(orders => console.tron.log('orders', orders))
      .catch(err => console.tron.log(err));
```

Import format for other modules


```js
  import * as FlyBuyCore from 'react-native-bildit-flybuy-core';
  import * as FlyBuyPickup from 'react-native-bildit-flybuy-pickup';
  import * as FlyBuyNotify from 'react-native-bildit-flybuy-notify';
  import * as FlyBuyPresence from 'react-native-bildit-flybuy-presence';
  import * as FlyBuyLiveStatus from 'react-native-bildit-flybuy-livestatus';
```

All function names are not modified, you just need to change `FlyBuy.<MODULENAME>` to one of the above import alias.
```

File: mono/packages/core/ios/RnFlybuyCore.h [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+- (void)getCurrentCustomer:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
+
+- (void)createCustomer:(NSDictionary *)customer
+    withResolver:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
+
+- (void)updateCustomer:(NSDictionary *)customer
+    withResolver:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
```
Comment (botnerd) [inline mono/packages/core/ios/RnFlybuyCore.h:39]: All of these methods belong under `FlyBuyCore.Customer`

File: mono/packages/core/ios/RnFlybuyCore.h [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+- (void)fetchSitesByRegion:(NSDictionary *)params
+    withResolver:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
+
+- (void)fetchSiteByPartnerIdentifier:(NSDictionary *)params
+    withResolver:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
```
Comment (botnerd) [inline mono/packages/core/ios/RnFlybuyCore.h:54]: All of these methods belong under `FlyBuyCore.Sites`

File: mono/packages/core/ios/RnFlybuyCore.h [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                              withState:(NSString *)state
+                              withSpot:(NSString *)spot
+                              withResolver:(RCTPromiseResolveBlock)resolve
+                           withRejecter:(RCTPromiseRejectBlock)reject;
+
+-(void)rateOrder:(NSInteger)orderId
+      withRating:(NSInteger)rating
+      withComments:(NSString *)comments
+      withResolver:(RCTPromiseResolveBlock)resolve
+    withRejecter:(RCTPromiseRejectBlock)reject;
```
Comment (botnerd) [inline mono/packages/core/ios/RnFlybuyCore.h:107]: All of these methods belong under `FlyBuyCore.Orders`

**Review by botnerd — CHANGES_REQUESTED** [review]
Please move the `FlyBuyCore` methods into groups (`Customer`, `Sites`, and `Orders`) like they were prior to the modularization. Everything else looks good.

**Review by botnerd — APPROVED** [review]
Everything looks good to me!

**Pull request merged**